patroni/async_executor.py

```diff
@@ -83,14 +128,32 @@ def schedule(self, action: str) -> Optional[str]:
 
     @property
     def scheduled_action(self) -> Optional[str]:
+        """The currently scheduled action, if any, else ``None``."""
         with self._scheduled_action_lock:
             return self._scheduled_action
 
     def reset_scheduled_action(self) -> None:
+        """Unschedule a previously scheduled action, if any.
+
+        .. note::
+            Must be called once the scheduled task finishes or is cancelled.
+        """
         with self._scheduled_action_lock:
             self._scheduled_action = None
 
     def run(self, func: Callable[..., Any], args: Tuple[Any, ...] = ()) -> Optional[bool]:
```

Comment:
I think that I did it wrong, the `func` could return anything, not necessarily `bool`, and the [docstring](https://github.com/zalando/patroni/pull/2704/files#diff-e79a66f0758d9f820b21814a9979d9067d4fa90b63ac1994376743a164f683aeR154-R155) confirms it.
```suggestion
    def run(self, func: Callable[..., Any], args: Tuple[Any, ...] = ()) -> Optional[Any]:
```

Reply:
Good catch!